libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1073,9 +1077,13 @@ void PbftManager::secondFinish_() {
 
 blk_hash_t PbftManager::generatePbftBlock(const blk_hash_t &prev_blk_hash, const blk_hash_t &anchor_hash,
                                           const blk_hash_t &order_hash) {
-  auto propose_period = pbft_chain_->getPbftChainSize() + 1;
+  const auto period = pbft_chain_->getPbftChainSize();
+  const auto reward_votes = vote_mgr_->updateRewardVotes(period);  // No cert votes in period 0
```

Comment:
what is going on here ? Why would call this function here ? You just want to get reward_votes(cert votes you have for previous round) from memory and not call updateRewardVotes, which is supposed to update period data

Reply:
Proposed PBFT block should include previous period cert votes (cert votes in finalized previous period PBFT block + reward votes queue) as many as node can.
Node propose a PBFT block and if don't save reward votes in DB here, node will lose the reward votes if node get restart